src/google-recaptcha-provider.tsx

```diff
@@ -12,14 +12,12 @@ interface IGoogleReCaptchaProviderProps {
 }
 
 export interface IGoogleReCaptchaConsumerProps {
-  executeRecaptcha?: (action?: string) => Promise<string>;
+  executeRecaptcha: (action?: string) => Promise<string>;
 }
 
 const GoogleReCaptchaContext = React.createContext<
   IGoogleReCaptchaConsumerProps
->({
-  // dummy default context;
-});
+>({} as IGoogleReCaptchaConsumerProps);
```

Comment:
I don't think casting an empty object to an interface is a good practice. Basically it's wrong, it gives the type system a false impression but in reality no implementation of the `executeRecaptcha` is implemented.

A proper solution here is to provide a version of the `executeRecaptcha` function as the default context value. That function should throw an error if it's ever been called. Something like:

```javascript
const GoogleReCaptchaContext = React.createContext<
  IGoogleReCaptchaConsumerProps	  IGoogleReCaptchaConsumerProps
>({	>({
executeRecaptcha: async (action?: string) => {
    throw Error('Execute recaptcha function not yet implemented');
 }});
```